File: framework/ecommerce-framework-dxa-module/src/main/java/com/sdl/ecommerce/dxa/controller/CartController.java
package com.sdl.ecommerce.dxa.controller;

import com.sdl.ecommerce.api.CartFactory;
import com.sdl.ecommerce.api.ECommerceException;
import com.sdl.ecommerce.api.ECommerceLinkResolver;
import com.sdl.ecommerce.api.model.Cart;
import com.sdl.ecommerce.dxa.model.CartWidget;
import com.sdl.webapp.common.api.content.ContentProviderException;
import com.sdl.webapp.common.api.model.MvcData;
import com.sdl.webapp.common.controller.BaseController;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import javax.servlet.http.HttpServletRequest;

/**
 * Cart Controller
 * Manage AJAX cart requests and the actual rendering of the cart itself.
 *
 * @author nic
 */
@Controller
@RequestMapping("/")
public class CartController extends BaseController {

    @Autowired(required = false)
    private CartFactory cartFactory;

    @Autowired
    private ECommerceLinkResolver linkResolver;

    /**
     * Add product to cart (AJAX request)
     * @param request
     * @param productId
     * @return  new cart amount
     * @throws ECommerceException
     */
    @RequestMapping(method = RequestMethod.GET, value = "/ajax/cart/addProduct/{productId}")
    public @ResponseBody
    String addProductToCart(HttpServletRequest request, @PathVariable String productId) throws ECommerceException {
        if ( this.cartFactory == null ) {
            return "0";
        }
        Cart cart = this.getCart(request);
        if ( cart == null ) {
            cart = this.cartFactory.createCart();
            this.storeCart(request, cart);
        }
        cart.addProduct(productId);
        return Integer.toString(cart.count());
    }

    /**
     * Remove product from cart (AJAX request)
     * @param request
     * @param productId
     * @return new cart amount
     * @throws ECommerceException
     */
    @RequestMapping(method = RequestMethod.GET, value = "/ajax/cart/removeProduct/{productId}")
    public @ResponseBody
    String removeProductToCart(HttpServletRequest request, @PathVariable String productId) throws ECommerceException {
        if ( this.cartFactory == null ) {
            return "0";
        }
        Cart cart = this.getCart(request);
        if ( cart == null ) {
            cart = this.cartFactory.createCart();
            this.storeCart(request, cart);
        }
        cart.removeProduct(productId);
        return Integer.toString(cart.count());
    }

    /**
     * Get stored cart from the HTTP session.
     *
     * @param request
     * @return cart
     */
    private Cart getCart(HttpServletRequest request) {
        return (Cart) request.getSession().getAttribute(Cart.CART_URI.toString());
    }

    /**
     * Store cart on the HTTP session.
     * @param request
     * @param cart
     */
    private void storeCart(HttpServletRequest request, Cart cart) {
        request.getSession().setAttribute(Cart.CART_URI.toString(), cart);
    }
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.DXA/Controllers/CartController.cs
using Sdl.Web.Common.Logging;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace SDL.ECommerce.DXA.Controllers
{
    /// <summary>
    /// Cart Controller
    /// </summary>
    public class CartController : System.Web.Mvc.Controller
    {
        /// <summary>
        /// Add Product to Cart
        /// </summary>
        /// <param name="productId"></param>
        /// <returns></returns>
        public ActionResult AddProductToCart(string productId)
        {
            var cart = ECommerceContext.Cart;
            if (cart == null )
            {
                cart = ECommerceContext.Client.CartService.CreateCart();
            }
            cart = ECommerceContext.Client.CartService.AddProductToCart(cart.Id, productId, 1);
            ECommerceContext.Cart = cart;
            return Content(cart.Count.ToString());
        }

        /// <summary>
        /// Remove Product from Cart
        /// </summary>
        /// <param name="productId"></param>
        /// <returns></returns>
        public ActionResult RemoveProductFromCart(string productId)
        {
            Log.Info("Removing product from cart: " + productId);
            var cart = ECommerceContext.Cart;
            if (cart == null)
            {
                cart = ECommerceContext.Client.CartService.CreateCart();
                return Content("0");
            }
            cart = ECommerceContext.Client.CartService.RemoveProductFromCart(cart.Id, productId);
            ECommerceContext.Cart = cart;
            return Content(cart.Count.ToString());
        }
    }
}